### ISSUE TITLE
Fix missing roles in wikidata/missing in members of parliament file

### ISSUE THREAD

**Issue by MansMeg** [welfare-state-analytics/riksdagen-corpus#252]
Some MPs are not included in the member_of_parliament.csv file. I guess they are missing the role in wikidata.

Hence we should probably systematically try to identify people with this lacking. An example:

Gustaf Johnsson, Q111804160,
https://www.wikidata.org/wiki/Q111804160
You can also find him in the "Biografibanden", Band 2, p. 285

**Comment by fredrik1984** [comment]
Maybe this is something to prioritize when we all MPs have a wiki-id? How was it, did Emil finish the work of going through the MP list of the bicameral period and check which ones that have a wiki-id?

**Comment by MansMeg** [comment]
Yes. Lets wait for Emils work. Then we can start this.

**Comment by fredrik1984** [comment]
I use issue this thread to report missing metadata for MPs in Wikidata.

This person https://www.wikidata.org/wiki/Q5969615 has no data for chamber (FK), party (bf = bondeförbundet), i-ort (Löfvander i Kvarnby). Source: bio book 3, p. 239.

**Comment by salgo60** [comment]
@fredrik1984 
* [X] [Q5969615](https://www.wikidata.org/w/index.php?title=Q5969615&action=history) ✅ 

as you understand Wikidata is not rocket science and people like me are more [bold](https://en.wikipedia.org/wiki/Wikipedia:Be_bold) --> we get some edit wars....
* I have done the below decisions
  * Try to use the bok [Tvåkammarriksdagen](https://www.wikidata.org/wiki/Q110346241) as the bible
  * Occupation is sadly not well structured in WD we have objects for them but as I have shown it is a small chaos out there [salgo60/HISCOKoder](https://github.com/salgo60/HISCOKoder)
  * Party
     * We get a small war in sv:Wikipedia if something is displayed with the wrong label (most people in sv:Wikipedia dont understands WD or just hate it ;-() 
     * for Swedish Church parish an user [Larske](https://sv.wikipedia.org/wiki/Special:Bidrag/Larske) has implemented that when we pick a Swedish church parish from Wikidata a check is done what was the name of the object at that time e.g. "[Gävle församling](https://www.wikidata.org/wiki/Q10512441)" has that stored names with start/end in [Q10512441#P2561](https://www.wikidata.org/wiki/Q10512441#P2561) 
         * _this is not implemented for parties_ and we wait on maybe you to be an authority telling what party names are the same parties and when they changed name and what should be split to more parties....
         * so I check in book Tvåkammar Riksdagen and if they say (bf) I have created a new object [Q110472693](https://www.wikidata.org/wiki/Q110472693) that has [P460](https://www.wikidata.org/wiki/Property:P460) as "said to be same as" --> the SPARQL will be more difficult BUT what is displayed in sv:Wikipedia [Axel Löfvander](https://sv.wikipedia.org/wiki/Axel_L%C3%B6fvander) and in Wikidata is "correct" and hopefully we get a better solution....
         * my opinion is that _Wikidata core_ lacks a function _give me this object with the correct name year 1934_ - **[WD backlog](https://phabricator.wikimedia.org/tag/wikidata/)**
      * Pictures are stored in WIkicommons that has its own "Wikidata" called [Wikibase](https://wikiba.se/) in a project structured data in commons this was implemented [Commons:Structured_data](https://commons.wikimedia.org/wiki/Commons:Structured_data) [video](https://www.youtube.com/watch?v=lmWmMIuCJVM) --> in Wikicommons we have [semantic interoperability](https://en.wikipedia.org/wiki/Semantic_interoperability) with Wikidata and sets **depicts Q5969615** --> search [haswbstatement:P180=Q5969615](https://commons.wikimedia.org/w/index.php?search=haswbstatement%3AP180%3DQ5969615&title=Special:Search&ns0=1&ns6=1&ns12=1&ns14=1&ns100=1&ns106=1)
         * e.g. picture with unique id [M130845966](https://commons.wikimedia.org/entity/M130845966) has json --> [M130845966.json](https://commons.wikimedia.org/entity/M130845966.json) --> **P180 depict Q5969615**         
         ```
         "statements": {
                  "P180": [
                       {
                           "mainsnak": {
                                   "snaktype": "value",
                                   "property": "P180",
                                   "hash": "95268a1b7ec81a7709cbf48f610db4ecb23067bb",
                                   "datavalue": {
                                             "value": {
                                             "entity-type": "item",
                                             "numeric-id": 5969615,
                                             "id": "Q5969615"
                                     },
                                    "type": "wikibase-entityid"
                            }
                         },
                         "type": "statement",
                          "id": "M130845966$F092D723-3261-46BD-9CDA-FC50AC340F69",
                          "rank": "preferred"
                       }
                  ]
         ```

**Comment by fredrik1984** [comment]
Thanks @salgo60 ! I appreciate your commitment!

**Comment by salgo60** [comment]
Its @Ainali @miroli  @tmtmtmtm @belteshassar @SchermanJ  who started it and as said we dont speak much with each other - interesting queries @miroll WD user Popperipop wrote down see [#112](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/112) can give an indication what he plans.....

* also I am inspired what you do **its the first time I see humaniora related metadata management** that is inline with how we worked when I was a scrum master on SEB handling international transactions... "with 0 tolerance for mistakes"  

FYI @fredrik1984 
* i started to look into
  *  salary / net worth [found in Census 1930 #103](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/103) - lesson learned not everyone is in this data...
  * [education levels from Census 1930 #103](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/105) -  not started
  * [archives from NAD #104](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/104) - I lack that Riksarkivet dont communicate [NAD roadmap 2023/2024 #3](https://github.com/Riksarkivet/dataplattform/issues/3) - done a POC miss a good dialogue with Riksarkivet
  * connecting Graves of Swedish PM people with Gravstensinventeringen [#22](https://github.com/salgo60/Gravstensinventeringen-Wikidata/issues/22) - starting
  * explained the Wikidata model how we track facts that are not confirmed and in Wikidata rank uncertain see [Riksarkivet SBL #36 related errors we think we have found](https://github.com/salgo60/Svenskaforsamlingar/issues/36) and also how we support more languages
  * how we use SPA to get more old pictures of Swedish PM see [#117](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/117)
  * how [chatGPT can be used to ask Wikidata with SPARQL #118](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/118)
  * [connect with "Hembyggsföreningar"](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/120)
 
 FYI:  
 * we are also using dashboards to check how much metadata we have about Swedish Parlament people see [Wikidata:WikiProject_every_politician/Sweden#Coverage](https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician/Sweden#Coverage)
  
  [<img width="1262" alt="image" src="https://user-images.githubusercontent.com/14206509/232754556-034c8911-ed2e-4ef6-bc2d-23ad31a3e267.png">](https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician/Sweden#Coverage) 
  
  * the same for [all Politicians in the world](https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician#Statistics)
    * scrapers for > [200 parliaments](https://github.com/every-politician-scrapers) -  [Swedish PM updates](https://github.com/every-politician-scrapers/sweden-riksdag-api-current/commits/main/)  done by @tmtmtmtm    
    
[<img width="1146" alt="image" src="https://user-images.githubusercontent.com/14206509/232754932-54d67b20-9142-420c-a60c-3d8c8033ad29.png">](https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician#Statistics)

**Comment by fredrik1984** [comment]
This person https://www.wikidata.org/wiki/Q5804642 miss i-ort (Heüman i Jönköping). Bio book 2, p. 159.

**Comment by fredrik1984** [comment]
This person https://www.wikidata.org/wiki/Q6190872 is missing some party metadata, see https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfQAQ

**Comment by salgo60** [comment]
@fredrik1984 thanks

* [X] [Q6190872](https://www.wikidata.org/wiki/Q6190872) ✅ 
* [X] [Q5804642](https://www.wikidata.org/wiki/Q5804642) ✅ 

FYI:
* errors I have seen
  * Liberalerna has been used for many people long before the name change... as said Wikidata is not rocket scientists meeting ;-) I also think we have no control of communists see video below when I play with the timeline....
  * one way to track issues is check the [Timeline query I have done](https://js.histropedia.com/apps/query-timeline/index.html?q=SELECT%20distinct%20?person%20?personLabel%20(SAMPLE(?SPA)%20AS%20?SPA)%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?partyLabel%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20?person%20wdt:P39%20?member;%0A%20%20%20%20wdt:P1343%20?source.%0A%20%20?person%20p:P1343%20?pTva.%0A%20%20OPTIONAL%7B?person%20wdt:P102%20?party%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%20%20?pTva%20ps:P1343%20wd:Q110346241.%0A%20%20?pTva%20prov:wasDerivedFrom%20%5B%20pr:P4819%20?SPAid%20%5D.%0A%0A%20%20OPTIONAL%20%7B?person%20wdt:P18%20?bild%7D.%0A%20%0A%20%20BIND(URI(CONCAT(%22https://portrattarkiv.se/details/%22,?SPAid))%20AS%20?SPA)%0A%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?partyLabel%0Aorder%20by%20?partyLabel&md=true&g=article&l=SPA&t=personLabel&s=birth&e=death&i=bild&d=0&c=partyLabel&f=partyLabel&v=t)
  
  * [video long about this ](https://youtu.be/-b51VFgaJcQ) in Swedish and also
     * Scholia - [Gymnasium Q55043](https://scholia.toolforge.org/topic/Q55043) - [Anna Lindh](https://scholia.toolforge.org/author/Q208591)
        * video chat with [Daniel Mietchen](https://orcid.org/0000-0001-9488-1870) - FIZ Karlsruhe - Leibniz-Institut fur Informationsinfrastruktur - 2020 if [Scholia could be customized for politician](https://www.youtube.com/live/yMtP64yFKn0?feature=share&t=7264)
           * in the video he mention [ordia](https://ordia.toolforge.org/)
     * how I connect pictures from SPA see also [#117 ](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/117)
     * using chatGPT to write Wikidata SPARQL [#118](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/118)
     * @dpriskorn tool to set main subject on Wikidata objects [ItemSubjector](https://github.com/dpriskorn/ItemSubjector)
     * [how to check all "talmän added etc."](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/119) / same but also [number of entries](https://w.wiki/6cCX)

**Comment by fredrik1984** [comment]
@salgo60 yes indeed, the hierarchical tree of "party" formations and splits in the bio book is fascinating. What constitutes a party before 1900 is not always clear. But in the Swerik project, we have also decided to use the bio books as our bible. However, in the end, we might merge some of the party names that represent the same party but with different names. 

We are currently working on a gold standard for name introduction for the whole period 1867–today, and that is why post some of the issues we come across in doing that work. Appreciate that you help out correcting stuff on Wikidata!

**Comment by salgo60** [comment]
@fredrik1984 you also have "vilde"

* in wikidata they suggests that "vilde" is an empty value for party but I have started to create WD objects for different types of "vilde"
  * example [SPARQL](https://w.wiki/6cBn)
  
[<img width="978" alt="image" src="https://user-images.githubusercontent.com/14206509/233414074-bdfe6815-d279-457c-9e7c-ff37d06767a6.png">](https://w.wiki/6cBn)

* [same list](https://w.wiki/6cC3) but also an example of a person and link the picture in SPA e.g. frisinnad vilde - [Kerstin Hesselgren](http://www.wikidata.org/entity/Q4953847) - [SPA](https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfStQ)


[<img width="978" alt="image" src="https://user-images.githubusercontent.com/14206509/233415238-ad385ebf-c6c9-4ef4-84f6-e2002fc53fdc.png">](https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfStQ)

[<img width="1131" alt="image" src="https://user-images.githubusercontent.com/14206509/233416094-fa700c4d-5035-4215-9075-0537ac71c0cf.png">](https://w.wiki/6cC3)

* example vänstervilde / vänstersocialistisk vilde
[<img width="594" alt="image" src="https://user-images.githubusercontent.com/14206509/233416886-017a253d-9c81-44e8-b1e9-4f2d66c5f790.png">](https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfN4g)

* all values we [have as party (also partigrupp)](https://w.wiki/6cCC) / same [query but also number of entries](https://w.wiki/6cCX)

[<img width="1301" alt="image" src="https://user-images.githubusercontent.com/14206509/233494012-e96d6794-d862-47c1-9549-0f1015198cac.png">](https://w.wiki/6cCX)

* this is an error as [Sveriges kommunistiska parti Q1208859](https://sv.wikipedia.org/wiki/Sveriges_kommunistiska_parti) is a disambiguation page 

[<img width="1358" alt="image" src="https://user-images.githubusercontent.com/14206509/233428450-ec896a37-b274-4784-b17e-9c08e0f25810.png">](https://sv.wikipedia.org/wiki/Sveriges_kommunistiska_parti)

**Comment by Ainali** [comment]
> * n wikidata they suggests that "vilde" is an empty value for party but I have started to create WD objects for different types of "vilde"

The items may be fine, but they should not be used to populate [member of political party (P102)](https://www.wikidata.org/wiki/Property:P102) or [parliamentary group (P4100)](https://www.wikidata.org/wiki/Property:P4100).

**Comment by salgo60** [comment]
> The items may be fine, but they should not be used to populate [member of political party (P102)](https://www.wikidata.org/wiki/Property:P102) or [parliamentary group (P4100)](https://www.wikidata.org/wiki/Property:P4100).

@Ainali 
They are right now in **exception to constraint** [P102#P2302](https://www.wikidata.org/wiki/Property:P102#P2302) which is a "good solution" until someone decide as @fredrik1984 states it "What constitutes a party before 1900 is not always clear" and we get a "golden standard" maybe we need new properties.... 

<img width="1030" alt="image" src="https://user-images.githubusercontent.com/14206509/233491430-677e4ee2-bfc8-479a-b199-d1c60846300d.png">

Olof Karswall wrote a paper "[Historical Settlement Units as Linked Open Data](https://ceur-ws.org/Vol-2364/24_paper.pdf)" I guess we need something like that for "political parties/parliamentgroupes/ active politician who has left the party they were elected for/ [independent politicians](https://en.wikipedia.org/wiki/Independent_politician).... from a project as [ParlaMint](https://www.clarin.eu/parlamint) that includes more countries ?!?!?!

[<img width="927" alt="image" src="https://user-images.githubusercontent.com/14206509/233493223-86280d09-d07b-4f9c-bcce-2b2e63ac5260.png">](https://ceur-ws.org/Vol-2364/24_paper.pdf)

**Comment by fredrik1984** [comment]
We will discuss the party issue before 1900 in the Swerik project. Back in that time "parties" were more lists or groups that voted for the same thing in the parliament. Although they were not parties in our modern definition I think it is reasonable to still tag them as parties. 

"Vilde" sounds like a good category to be included, especially since our "bible" (the bio books) uses that term.

**Comment by salgo60** [comment]
My understanding is that @Ainali and @miroli are doing it right now as the rest of the Wikidata community e.g.

* Amineh Kakabaveh [Q3675519](https://www.wikidata.org/wiki/Q3675519)
  

<img width="1308" alt="image" src="https://user-images.githubusercontent.com/14206509/233542340-d224d9ce-c919-426d-812f-5fc74ba6cd39.png">


  * in WIkidata we also have an construction [end cause P1534](https://www.wikidata.org/wiki/Property:P1534) 
     * if we look at Kakabaveh we should in my opinion have a WD object [end cause P1534](https://www.wikidata.org/wiki/Property:P1534)  in Swedish sv:Wikipedia says "_Den 28 augusti 2019 beslutade Vänsterpartiets verkställande utskott att inleda ett uteslutningsärende gällande Kakabaveh.[[24]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-:0-24)[[25]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-v-r-20190828-25) Detta på grund av att hon, enligt partisekreteraren [Aron Etzler](https://sv.wikipedia.org/wiki/Aron_Etzler), "ofta och allvarligt skadar förtroendet för partiet gentemot väljarna och mot sina kollegor".[[24]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-:0-24) Kakabaveh anklagades bland annat för att inte ha närvarat vid statsministeromröstningen 2018,[[26]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-26) en omröstning där Vänsterpartiet lade ned sina röster.[[27]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-27) Hon anklagades även för att ha underlåtit att betala [partiskatt](https://sv.wikipedia.org/wiki/Partiskatt).[[28]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-28)_

_Kakabaveh hävdade att hon inte meddelades i förväg om uteslutningen utan fick reda på det via media.[[29]](https://sv.wikipedia.org/wiki/Amineh_Kakabaveh#cite_note-29) Hon lämnade partiet samma dag på egen begäran_

* Emma Carlsson Löfdahl WD [Q4941925](https://www.wikidata.org/wiki/Q4941925)

   * [Q4941925#P102](https://www.wikidata.org/wiki/Q4941925#P102)  
    
   <img width="1104" alt="image" src="https://user-images.githubusercontent.com/14206509/233545398-5df4a81e-c98a-434a-b477-96ba8c5d2f66.png">
   
   ----

* end cause are mostly used for [position held P39](https://www.wikidata.org/wiki/Property:P39) is my understanding but as I say above maybe it should be used for parties also.... **structure is always good ;-)**
   * [SPARQL](https://w.wiki/6cJT) for Sweden
   
  [<img width="1377" alt="image" src="https://user-images.githubusercontent.com/14206509/233548473-7659ce3c-66c8-444e-8d92-977d19f78081.png">](https://w.wiki/6cJT)
  
  * my vote is also that a project like [ParlaMint](https://github.com/clarin-eric/ParlaMint) has a controlled "vocabulary" for end causes... so we get some kind of consistency if not ParlaMint is possible a start is doing it in a project "Välfärden analyseras". When I look at [ELI](https://op.europa.eu/documents/2050822/2138819/ELI+Implementation+Methodology.pdf/#:~:text=The%20ELI%20approach%20aims%20to,unique%20legislative%20and%20legal%20tradition.) they suggest usage of controlled vocabularies... which I think is a good pattern....
 ___ 
* [Riksdagen får sin andra vilde – tangerar rekord](https://www.altinget.se/rikspolitik/artikel/riksdagen-faar-sin-andra-vilde-tangerar-rekord)

**Comment by salgo60** [comment]
@fredrik1984 another research "wet dream" I guess is the connection between Valmyndigheten and the Swedish PM data see my try 2019 [#85 Valmyndigheten koppling Riksdagens öppna data](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/85) maybe you can get that data as researcher I hope they have the Swedish "personnummer" in both systems but I didnt get a good license

Looks like UK has coordinates for election districts in [Wikidata map below](https://w.wiki/BJ2) see example district [Q3238840](https://www.wikidata.org/wiki/Q3238840)

[<img width="1065" alt="image" src="https://user-images.githubusercontent.com/14206509/233591337-c47cc5f9-efa2-43ba-8d07-598528809ebc.png">](https://w.wiki/BJ2)

OT in Wikidata we supports areas - shapes - which is even better than a singel coordinate --> would be nice to get electoral Swedish districts  as shapes
* see Nature reserve where "Naturvårdsverket" has given us CC-0 shapes for Nature reserves in Sweden...
   * [ SPARQL Nature reserves in Umeå, Sweden](https://w.wiki/6cWu)
      * example Ormberget-Hertsölandet WD [Q30174267](https://www.wikidata.org/wiki/Q30174267) has shape [Ormberget Hertsölandet/2021406.map](https://commons.wikimedia.org/wiki/Data:/Sweden/Nature_reserves/2020/Ormberget_Herts%C3%B6landet/2021406.map)
         
[<img width="1010" alt="image" src="https://user-images.githubusercontent.com/14206509/233757000-23f94662-7fee-4cfc-9f51-d9b75f8f7046.png">](https://w.wiki/6cVT)

**Comment by salgo60** [comment]
@fredrik1984 Another related discussion on sv:Wikipedia regarding the names of parties and they the need to have  the correct name when something happened "[Wikipedia:Bybrunnen#Inkonsekventa_partinamn](https://sv.wikipedia.org/wiki/Wikipedia:Bybrunnen#Inkonsekventa_partinamn)"

1. Wikipedia lacks an authority 
2. Easy get this data as CC-0 and machine readable

**Comment by fredrik1984** [comment]
https://www.wikidata.org/wiki/Q6178299 is missing i-ort (Sjödahl i Göteborg) and specific start dates in FK (20 March 1931). Source: bio book 4, p. 150. 

He has not got a link to the Swedish portrait archive either.

**Comment by fredrik1984** [comment]
https://www.wikidata.org/wiki/Q6012010 is missing i-ort (Nilsson i Mölndal). Source: bio book 4, p. 128.

He has not got a link to the Swedish portrait archive.

**Comment by BobBorges** [comment]
Re the original issue, is this (image) what's missing that causes the people/wiki_ids to not get scraped into our metadata?

![image](https://user-images.githubusercontent.com/8887511/233922736-5267a43c-585a-4a4d-9bf3-1983a1b72e8d.png)

It's related to issue #265 and I would / could just run through our known_mps/catalog and add this attribute to the ones who don't have it.

**Comment by MansMeg** [comment]
@BobBorges Im not sure, but I dont think so. @ninpnin knows the script to update the mp database based on wikidata. I would check how they are selected in the API query to wikidata. I think they are selected based on role.

**Comment by BobBorges** [comment]
So, it's these:

![image](https://user-images.githubusercontent.com/8887511/233937286-df309fe1-9a72-4701-b8c7-00e75ad8bf25.png)

wd:Q10655178 
wd:Q33071890 
wd:Q81531912

**Comment by MansMeg** [comment]
I guess so, but @ninpnin should confirm. Are the MPs missing in our database missing these attributes?

**Comment by BobBorges** [comment]
We talked about it last week -- if I understand what's going on correctly -- this causes missing_member_of_parliament, and also some of the other `missing_`s are because without this attr the wiki_id doesn't make it into the query results in the first place.

**Comment by MansMeg** [comment]
Probably. Another argument why to try to fix the tests sequentially. ie. start with fixing so the person.csv test pass. Then we merge that test suite and the updated mp database that passes the first test. Then take the next test etc.

**Comment by BobBorges** [comment]
missing member and missing person are both 103 IDs and the rows in the summary (print(df)) are the same – if we fix member attr on wikidata, they might both resolve at once

**Comment by salgo60** [comment]
@BobBorges et al do you have a good english translation of the book title "Enkammarriksdagen 1971-1993/94" same as [Q111443541](https://www.wikidata.org/wiki/Q111443541) right now we miss an english label. If you have just update [WD](https://www.wikidata.org/wiki/Q111443541)

[<img width="863" alt="image" src="https://user-images.githubusercontent.com/14206509/233947764-825ac33d-2676-40b8-84ce-3db6e9f80073.png">](https://www.wikidata.org/wiki/Q111443541)

**Comment by BobBorges** [comment]
I guess it would be _Unicameral Parliament_.

**Comment by fredrik1984** [comment]
I think a more correct translation would be Unicameral Riksdag. 
[flersprakig-ordlista-nov2020.pdf](https://github.com/welfare-state-analytics/riksdagen-corpus/files/11308637/flersprakig-ordlista-nov2020.pdf)


A good Swedish-English dictionary with Swedish parliamentary terms is the one attached here.

**Comment by BobBorges** [comment]
Ok Here's a list of Wiki IDs causing the member_of_parliament unit test to fail -- it means they're missing the role wd:Q10655178, wd:Q33071890, or wd:Q81531912.


- [ ] Q5581563 | Bonde i Fållnäs
- [ ] Q19975949 | Ekman i Jönköping senare Stocksund
- [ ] Q116199773 | Gilljam i Stockholm
- [ ] Q5771035 | Grenholm i Stockholm
- [ ] Q5867622 | Gripenstedt i Nynäs senare Revlinge
- [ ] Q117035450 | Gustavson i Sala senare Västerås
- [ ] Q5779431 | Gyllenstierna af Lundholm i Bjärsgård
- [ ] Q5782356 | Hagerman i Stockholm
- [ ] Q99006280 | Hagman i Linköping
- [ ] Q117085828 | Hansson i Örebro
- [ ] Q116690088 | Hansson i Rättvik
- [ ] Q5789203 | Hansson i Klovsta
- [ ] Q117221213 | Hansson i Ormastorp
- [ ] Q117221276 | Hansson i Vallberga
- [ ] Q5789670 | Hansson i Torp
- [ ] Q5797392 | Heidenberg i Visby
- [ ] Q5797565 | von Heijne-Lillienberg i Edeby
- [ ] Q5801387 | Henriksson i Laholm
- [ ] Q5801406 | Henriksson i Kolbuxeröd
- [ ] Q5805033 | Hierta i Näsby
- [ ] Q117221366 | Huss i Ekhult
- [ ] Q5818643 | Hygrell i Lycksele
- [ ] Q5823030 | Indebetou i Forssa
- [ ] Q5824384 | Isaksson i Nygården
- [ ] Q5824491 | Isberg i Norrtälje
- [ ] Q5885281 | Johansson i Fägerhult
- [ ] Q117221619 | Johansson i Rimforsa
- [ ] Q117222379 | Johansson i Onsjö senare Larv
- [ ] Q116687298 | Johansson i Bronäs
- [ ] Q6140081 | Johnsson i Norr-Kärra
- [ ] Q5889587 | Johnsson i Remmarlöv
- [ ] Q117222503 | Jonasson i Gullaboås
- [ ] Q5891393 | Jonsson i Sundsbyn
- [ ] Q5892646) | Josefsson i Halmstad
- [ ] Q5926472 | Kunze i Göteborg
- [ ] Q29034014 | Lamberg i Ystad
- [ ] Q40759724 | von der Lancken i Kristianstad
- [ ] Q5937483 | Larsson i Rönnsbol
- [ ] Q5956515 | Lindh i Säffle
- [ ] Q111916168 | Lindskog i Billinge
- [ ] Q5966083 | Lundblad i Eskilstuna
- [ ] Q5966794 | Lundgren i Göteborg
- [ ] Q5967728 | Lundström i Piteå
- [ ] Q5969231 | Låftman i Billingsfors
- [ ] Q5969775 | Lönnberg i Linköping
- [ ] Q5985428 | Metzén i Skänninge
- [ ] Q108333808 | Montgomery-Cederhielm i Segersjö
- [ ] Q6010969 | Nilsson i Viby
- [ ] Q98393527 | Nordenskjöld i Sjögelö
- [ ] Q112036699 | Norrby i Åkersberga
- [ ] Q111728193 | Ohlsson i Järrestad senare Gladsax o Simrishamn
- [ ] Q6042591 | Persson i Tofta
- [ ] Q6160772 | Persson i Vä
- [ ] Q6043249 | Persson i Häljebol
- [ ] Q117039047 | Petersson i Gärestad
- [ ] Q6161303 | Petersson i Möllstorp
- [ ] Q2086269 | von Platen i Vettinge
- [ ] Q6056573 | Printzensköld i Jönköping
- [ ] Q114729447 | Risberg i Kållered
- [ ] Q6078545 | Rooth i Helsingborg
- [ ] Q6082517 | Rudberg i Halltorp
- [ ] Q6084488 | Ruthberg i Grisbacka
- [ ] Q6085142 | Rydberg i Korsgården
- [ ] Q6166587 | Rönblad i Härnösand
- [ ] Q6086793 | Rönnberg i Stocked
- [ ] Q6166851 | Sahlström i Utterbyn
- [ ] Q6171696 | Segerlind i Uddersrud
- [ ] Q6177454 | Sjöcrona i Helsingborg
- [ ] Q6168498 | Schmidt i Partille
- [ ] Q6169091 | von Schoultz i Göteborg
- [ ] Q6178395 | Sjögreen i Mariannelund
- [ ] Q6178565 | Sjögren i Lund
- [ ] Q6178560 | Sjögren i Asarum
- [ ] Q6179348 | Skagerberg i Annefors
- [ ] Q117018060 | Smedh i Bergkarlås
- [ ] Q6188899 | Stenbäck i Kullinge
- [ ] Q6188945 | Stendahl i Eldsberga
- [ ] Q6190734 | Stjernstedt i Tisslinge
- [ ] Q6190864 | Stockenberg i Halmstad
- [ ] Q6194030 | Styffe i Stockholm
- [ ] Q6194477 | Stånggren i Linköping
- [ ] Q6331198 | Sundberg i Haparanda
- [ ] Q6199804 | Swartz i Norrköping
- [ ] Q6201341 | Säve i Karlstad
- [ ] Q6205904 | Tenger i Västervik
- [ ] Q6208993 | Thorburn i Uddevalla
- [ ] Q116146163 | Thörnberg i Ronneby
- [ ] Q48704901 | af Trolle i Karlskrona
- [ ] Q6216096 | Trägårdh i Ystad
- [ ] Q6216492 | Tufwesson i Lur
- [ ] Q6218905 | Uhlén i Malmö
- [ ] Q6228885 | Wahlgren i Stora Dicka
- [ ] Q6229360 | Walberg i Alingsås
- [ ] Q6230443 | Wallén i Nyebro
- [ ] Q27996966 | Welin i Herrljunga
- [ ] Q6237369 | Westblad i Husby
- [ ] Q6237483 | Westerberg i Falköping
- [ ] Q94645211 | Wikström i Karlskrona
- [ ] Q6243918 | Winborg i Stockholm
- [ ] Q6245359 | Witt i Helsingborg
- [ ] Q112133202 | Åkerström i Clemensnäs
- [ ] Q26211952 | Persson i Bromma
- [ ] Q117426463 | Rydén i Växjö

**Comment by MansMeg** [comment]
Great! @salgo60 Do you have time to look at these MPs?

**Comment by salgo60** [comment]
@BobBorges I cant check your list (guess I miss permissions) so I do it on [#121](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/121)

**Comment by fredrik1984** [comment]
https://www.wikidata.org/wiki/Q55955 is missing AK start date (16 March 1897). Source: https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfN4g

**Comment by salgo60** [comment]
@fredrik1984 FYI a discussion is now on [Wikipedia Bybrunnen](https://sv.wikipedia.org/wiki/Wikipedia:Bybrunnen/Arkiv_2023-05#Inkonsekventa_partinamn) that we are not consequent in the usage of the names of Swedish parties ... I suggest that we instead of having just tables in articles make this data as data in Wikidata or any other database...
>  I våra artiklar om äldre svenska städer, köpingar och kommuner (till exempel artikeln [Strömstads stad](https://sv.wikipedia.org/wiki/Str%C3%B6mstads_stad)) är vi inte konsekventa vad gäller partinamnen
1) best would be if you had this data... and we could quote you 🚀 I guess

[<img width="1096" alt="image" src="https://user-images.githubusercontent.com/14206509/234036133-0d205e01-8e95-4c0e-b6f8-a6671d2a7379.png">](https://sv.wikipedia.org/wiki/Str%C3%B6mstads_stad)

**Comment by fredrik1984** [comment]
https://www.wikidata.org/wiki/Q6209073 as a slightly incorrect i-ort (correct should be Thorell i Stolp-Ekeby). Also, he is missing party for https://www.wikidata.org/wiki/Q10554125. Source: bio book 1, p. 271. https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABHgKQ

**Comment by fredrik1984** [comment]
https://www.wikidata.org/wiki/Q6078273 is missing i-ort (Roos i Malmö). Source: bio book 3, p. 266.

**Comment by fredrik1984** [comment]
@salgo60 from Swerik's perspective, if a party just change name (e.g. from bondeförbundet to centerpartiet) then it should be regarded as the same party. But bonderförbundet is not the same as jordbrukarnas riksförbund (the latter merged into the former in 1921).

**Comment by MansMeg** [comment]
Are you sure about this Fredrik? I think we might want an instance per name so folkpartiet becomes liberalerna. Even though the party as such has not changed, the data has. So we need to keep track of all the names (and their abbreviations) to map to the protocols, is my guess.

**Comment by fredrik1984** [comment]
I guess my comment was more from a historical perspective. From a technical/curational point of view I would say you are right @MansMeg

**Comment by MansMeg** [comment]
Yes. When I think about it this is not very clear how to define. We could also say that we have multiple instances of a name for the same party.

**Comment by BobBorges** [comment]
@salgo60 how confident are you in the iorts from the list you and @Emil produced? If you're relatively confident they're mostly OK, I will add them all to wikidata programatically which would sort out @fredrik1984 's comment about Roos i Malmö and the other 500ish MPs missing an iort.

**Comment by MansMeg** [comment]
Should we wait to update all these iorts? Im a little hesitant to bulk upload before we have done any quality control of the files.

**Comment by BobBorges** [comment]
This is why I was asking about it. I was under the impression that the list I got from  Emil (why can't I tag him btw?) was hand curated from the bio books -- if that's the case it should be pretty good. If it's not the case then ¯\\\_⦅ツ⦆\_/¯ we should quality control it.

**Comment by salgo60** [comment]
> https://www.wikidata.org/wiki/Q6012010

@fredrik1984 **done** ✅ 
* interesting how much information [Möndalsposten](https://www.molndalsposten.se/folk/theodor-nilsson-var-toppolitiker-for-100-ar-sedan-nadde-anda-till-riksdagen.56fe1e9f-5385-4554-b2f6-411e254b50f7) has about him

**Comment by salgo60** [comment]
> @salgo60 how confident are you in the iorts from the list you and @emil produced? If you're relatively confident they're mostly OK, I will add them all to wikidata programatically which would sort out @fredrik1984 's comment about Roos i Malmö and the other 500ish MPs missing an iort.

@BobBorges 
The only error so far is
* [#121 WD Q116199773](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/121#issuecomment-1520407933)
* plus a spelling mistake from the pdf,,,

If you do the update as an [transaction](https://github.com/welfare-state-analytics/riksdagen-corpus/pull/278#issuecomment-1519766846) then we could rollback if it looks too crazy.....
* I plan later add more things from Emils files for the Enkammar...book if no one else does it....

**Question**: What does iort stands for?

**Comment by fredrik1984** [comment]
@salgo60 iort/i-ort stands for "i riksdagen kallad", used by the speaker of the house to address MPs, often the place where an MP lived. I just found this wiki page: https://sv.wikipedia.org/wiki/I_riksdagen_kallad

Good thing to know that after 1976/77 they stopped address MPs with herr/fru/fröken. "Sedan 1977/1978 betecknas ledamöterna istället med både för- och efternamn, men vid behov av särskiljning tillfogas ett orts- eller gårdsnamn"

**Comment by fredrik1984** [comment]
https://www.wikidata.org/wiki/Q6012769 is missing i-ort (Nisser i Grycksbo) and specific start date in FK (11 January 1938). He is also missing a reference to the portrait archive. Source: bio book 5, p. 80. ping @salgo60 (do you want me to ping you for each post of a missing MP metadata in Wikidata?)

A question to @MansMeg @BobBorges @ninpnin – regarding Bob's comment above, should I not continue reporting these missing MP metadata on Wikidata? I thought that Emil had already added all missing MPs on Wikidata, but that some metadata is still missing, hence why I add these comments.

**Comment by MansMeg** [comment]
Please add this Fredrik. Emil and @salgo60 added all people on wikidata that was missing. Although not all metadata has been added. We do this iteratively. So now we have found people that is missing information on role that needs to be fixed so we get all mps in. Thats the first step. Then we gonna add more and more metadata on a need to have basis.

**Comment by fredrik1984** [comment]
Ok, good, I thought so! Going through Väinö's CSV file with MP introduction is very good to see what the quality of the MP metadata in Wikidata looks like, and it is often very good. 

@Lottabrorsson – would be great if you and Mattias also report missing MP metadata in Wikidata here when you go through your share of MPs in the CSV file!

**Comment by fredrik1984** [comment]
@ninpnin – this FK-MP (1942–1945) https://www.wikidata.org/wiki/Q6175985 is missing from the input/mp/fk files: https://github.com/welfare-state-analytics/riksdagen-corpus/tree/main/input/mp/fk

https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfNlg

**Comment by salgo60** [comment]
@fredrik1984 WD is behind... my understanding was that Emil added often the Alias and some data if it was empty...

I try to add
1. Name with form "I riksdagen kallad"
2. The link to SPA  (OT I checked the number of views we have in WIkidata on pictures from SPA and its > 200 000 views per month and pictures used on 74 different Wikis... **nice scaling** [link](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/117#issuecomment-1514422898))
3. Try to set first names
4. party: try to follow the Bible but I am not skilled and do mistakes...
5. birth/ death location and date and also check if we have other sources and somtimes add the churchbook
6. does a search on Svenska Gravar and add [Gravstensinventeringen](https://github.com/salgo60/Gravstensinventeringen-Wikidata/issues/22)
7. Check with the gadget [Authority Control](https://www.wikidata.org/wiki/Q5779431#) if the person is in VIAF/LIBRIS XL....
8. updates the sv:Wikipedia article with the name "I riksdagen kallad" and if no template is used I add [Faktamall biografi WD](https://sv.wikipedia.org/wiki/Mall:Faktamall_biografi_WD) and [Auktoritetsdata](https://sv.wikipedia.org/wiki/Mall:Auktoritetsdata)- see [change Axel Roos](https://sv.wikipedia.org/w/index.php?title=Axel_Roos_(politiker)&diff=prev&oldid=52589480)
9. [Add pictures from SPA](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/117) and metadata depict WD object
10. "Described by source P1343" - [Tvåkammar-riksdagen 1867–1970](https://www.wikidata.org/wiki/Q110346241) / Q110346241
--> We now have > [3400 First/Sec chamber people with this source](https://w.wiki/6d63) / as a [table](https://w.wiki/6dE3)


[<img width="1420" alt="image" src="https://user-images.githubusercontent.com/14206509/234193249-1d26dd48-e0f4-47f6-92b9-d408317efe0a.png">](https://w.wiki/6d63)

**Comment by salgo60** [comment]
> ping @salgo60 (do you want me to ping you for each post of a missing MP metadata in Wikidata?)

@fredrik1984 
Maybe better add a new issue in [salgo60/Wikidata_riksdagen-corpus](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues) and assign me or add to one of [my todo lists like #121](salgo60/Wikidata_riksdagen-corpus)

* [x] touched [William_Nisser](https://sv.wikipedia.org/wiki/William_Nisser_(1882%E2%80%931960))

**Comment by salgo60** [comment]
> @salgo60 iort/i-ort stands for "i riksdagen kallad", used by the speaker of the house to address MPs, often the place where an MP lived. I just found this wiki page: https://sv.wikipedia.org/wiki/I_riksdagen_kallad
> 
> Good thing to know that after 1976/77 they stopped address MPs with herr/fru/fröken. "Sedan 1977/1978 betecknas ledamöterna istället med både för- och efternamn, men vid behov av särskiljning tillfogas ett orts- eller gårdsnamn"

Thanks
:smiley: I was part of creating that Wikipage

* [x] I added iort and i-ort as alias for the WD object [Q110382440](https://www.wikidata.org/wiki/Q110382440) 
   * but I still doesnt understand how iort is related with "i riksdagen kallad" 😕 

[<img width="1136" alt="image" src="https://user-images.githubusercontent.com/14206509/234363136-eee07702-37e9-4c8c-9a31-f6f97ba5fdeb.png">](https://www.wikidata.org/wiki/Q110382440)

**Comment by fredrik1984** [comment]
Ok! I will start a new issue there and move the missing MP metadata in Wikidata things. Would it be possible for you @salgo60 to fix the rest of my MP posts in this thread?

**Comment by salgo60** [comment]
@fredrik1984 
@ping me when something is done right now I am confused were the latest version of your files are

In my version I found an error.... correct is 

[Q6157121](https://www.wikidata.org/wiki/Q6157121),[Nisser i Falun, Ernst M W, 5:79](https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfbtg)

**Comment by salgo60** [comment]
> @salgo60 from Swerik's perspective, if a party just change name (e.g. from bondeförbundet to centerpartiet) then it should be regarded as the same party. But bonderförbundet is not the same as jordbrukarnas riksförbund (the latter merged into the former in 1921).

@fredrik1984 @MansMeg 
I suggest that you start look into [SKOS](https://www.w3.org/2004/02/skos/) see my [try to explain for the people at Skansen](https://github.com/salgo60/SkansenWikidata/issues/11)

[<img width="1188" alt="image" src="https://user-images.githubusercontent.com/14206509/234602593-5c795922-f3d8-467e-a113-73bd23108c26.png">](https://github.com/salgo60/SkansenWikidata/issues/11)

* [owl:same as](https://www.w3.org/TR/owl-ref/#sameAs-def)
* [When owl:sameAs isn’t the Same: An Analysis of Identity Links on the Semantic Web](https://www.w3.org/2009/12/rdf-ws/papers/ws21)

I think as more mature you get on using linked data and more people get involved then you will split into more objects

* Riksarkivet Tora used to store a "socken" and a "församling" in the same object
   * Wikidata has one for  "Söderala församling" [Q10688470#P4820](https://www.wikidata.org/wiki/Q10688470#P4820) 
   and one for "Söderala socken"
      * below Riksarkivet use [rdfs:seeAlso](https://www.w3.org/wiki/UsingSeeAlso) --> ""that might provide additional information about the subject resource."
         *  I would say it would be better to use [SKOS](https://www.wikidata.org/wiki/Property:P4390) and a relation like "[broad match](https://www.wikidata.org/wiki/Q39894595)" or "[narrow match](https://www.wikidata.org/wiki/Q39893967)"
         * same problems with [LIBRISXL 2019 and August Strindberg](https://web.archive.org/web/20210403032657/https://kundo.se/org/librisxl/d/semantisk-koppling-i-librisxl/)

[<img width="1199" alt="image" src="https://user-images.githubusercontent.com/14206509/234620344-02a087f6-d6fa-4838-90d1-3b68007ffb3d.png">](https://tora.entryscape.net/snorql/?describe=https%3A%2F%2Fdata.riksarkivet.se%2Ftora%2Fmodernparish/218205000)
 
 *  **Another example** I did in 2023 jan a "split" of Wikidata Swedish church into church and cemetery see [Gravstensinventeringen-Wikidata](https://github.com/salgo60/Gravstensinventeringen-Wikidata)

**Comment by salgo60** [comment]
> @fredrik1984 you also have "vilde"
> 
> * in wikidata they suggests that "vilde" is an empty value for party but I have started to create WD objects for different types of "vilde"
>   
>   * example [SPARQL](https://w.wiki/6cBn)

@fredrik1984 as WD now has more pictures scanned of Swedish PM I did a test to filter out people described as "vilde" see [#139](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/139)
